tests/neg/i2473.scala

```diff
@@ -0,0 +1,7 @@
+trait Foo { trait Inner }
+trait Bar { foo: Foo =>
+  type Inner <: foo.Inner // error: type Inner cannot have the same name as trait Inner in trait Foo -- class definitions cannot be overridden
```

Comment:
I changed the message to print 
```scala
-- [E068] Syntax Error: tests/neg/i2473.scala:3:7 ------------------------------
3 |  type Inner <: foo.Inner // error
  |       ^
  |type Inner cannot have the same name as trait Inner in trait Foo -- cannot define type member with the same name as a trait member in self reference foo.
  |(Note: this can be resolved by using another name)
-- [E068] Syntax Error: tests/neg/i2473.scala:6:8 ------------------------------
6 |  class Inner // error
  |        ^
  |class Inner cannot have the same name as trait Inner in trait Foo -- cannot define class member with the same name as a trait member in self reference baz.
  |(Note: this can be resolved by using another name)
```